libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -20,12 +20,44 @@
 #include <libyul/AsmData.h>
 #include <libyul/backends/evm/EVMDialect.h>
 
+#include <libsolutil/Algorithms.h>
+
 using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 using namespace std;
 
-ControlFlowBuilder::ControlFlowBuilder(CFG::NodeContainer& _nodeContainer, FunctionFlow const& _functionFlow):
+namespace
+{
+
+bool exitReachable(FunctionFlow const& _functionFlow)
+{
+	bool foundExit = false;
+
+	BreadthFirstSearch<CFGNode const*>{{_functionFlow.entry}}.
+		run([&](CFGNode const* _node, auto&& _addChild) {
```

Comment:
Please use references unless it can be null.

Reply:
References can only be used if it is a comparable type which `CFGNode` is not